artiq/applets/big_number.py

```diff
@@ -55,7 +57,8 @@ def start_edit(self):
 
     def confirm_edit(self):
         value = float(self.edit_widget.text())
```

Comment:
be careful here, the `short_format` function may scale the value depending on the units provided, and when editing this needs to be handled instead of just parsing as float.